Discriminator value could be an integer

prohalexey opened the pull request:
The same as
```
399:    $values[] = $conn->quote((string) $subclassMetadata->discriminatorValue);
```

Without casting a TypeError is raised.

greg0ire commented:
The change on line 399 was introduced in a merge up: 5a40b99e11a0de532eb866f062dccad20da19add :thinking:

W0rma commented:
@prohalexey Does this PR fix https://github.com/doctrine/orm/issues/11341?

Does it make sense to add the unit tests which are suggested in this issue?

greg0ire commented:
> Does it make sense to add the unit tests which are suggested in this issue?

Yes, please do :pray:

prohalexey commented:
Any news here @greg0ire @W0rma ?

greg0ire commented:
@prohalexey can you please incorporate the unit tests suggested in #11341 ?

prohalexey commented:
Tests have been added and the code has been changed to a more accurate version. @greg0ire pls, look

prohalexey commented:
@greg0ire @SenseException @derrabus could someone check this pr ?

greg0ire commented:
Please kindly squash your commits together. If you don't, we'll try to remember to do it for you but it's best if you save us this trouble.

How to do that?

1. `git rebase -i origin/3.1.x`, assuming `origin` is a git remote that points to this repository, and not your fork. If you're not sure what your remotes are, run `git remote -vvv`, there should be your fork and the holy/reference/base/origin/whatever-you-call-it repository.
2. A window will show up with many lines, replace `pick` with `fixup` on every line but the first one
3. Close your editor, git should do its magic, and you should end up with one commit
4. Use `git push --force` to overwrite what you already push. Don't forget the `--force` option otherwise git will try to merge both things together.

Also, once you have a single commit, please improve your commit message according to [the contributing guide](https://www.doctrine-project.org/contribute/index.html#working-on-topic-branches). "Discriminator value could be an integer (TypeError is raised)" is not enough IMO.

prohalexey committed:
Using an integer as discriminator value with ORM v3

This fixes a bug that occurred when configuring integers as discriminator values. Doctrine throws a type error whenever the application generates queries.

prohalexey commented:
@greg0ire I think I'm done.

greg0ire commented:
Thanks @prohalexey !

yakijavier commented:
Hello @greg0ire , if it is supposed to be possible to use an integer as a descriminator, in the SqlWalker.php file in line 2255 shouldn't the discriminator be cast to string so that quote() accepts it?

greg0ire commented:
I think so yeah… my IDE prints a warning, but I'm not sure why static analysis does not report it. @prohalexey what do you think?

prohalexey commented:
I ran tests(GH11341Test, GH11199Test) and did not stop at this line (I set breakpoint at 2255).

prohalexey commented:
New PR https://github.com/doctrine/orm/pull/11456
